torch/testing/_internal/common_methods_invocations.py

```diff
@@ -204,6 +205,16 @@ def __init__(self,
         self.aten_name = aten_name if aten_name is not None else name
         self.variant_test_name = variant_test_name
 
+        # Attribute to verify dynamic_dtypes are used.
+        self.dynamic_dtypes = any(map(lambda dtypes: isinstance(
+            dtypes, opinfo_helper._dynamic_dispatch_dtypes), (dtypes, dtypesIfCUDA)))
+
+        if self.dynamic_dtypes:
```

Comment:
Would you elaborate on what this block of code does? Don't we already have a block that guarantees all the dtypes in the tuple are actually dtypes?

Reply:
This block is to make sure that if you have asked for dynamic dtypes then you have asked them for both `dtypes` and `dtypesIfCUDA`. 

If the user does,
```python
OpInfo('op',
    dtypes=get_supported_dtypes(..),
    samples=samples)
```

Then due to the way we set dtypes for other devices that are not present, we will set the same dtypes for CUDA.
So this check makes sure that such, OpInfo is ill-formed and errors out till the user acquires dynamic dtypes for `dtypes` and `dtypesIfCUDA`

https://github.com/pytorch/pytorch/blob/cab65ea3b94691af4e0ba530fa4aa070bc5c548e/torch/testing/_internal/common_methods_invocations.py#L249-L252

Reply:
Let's add a comment explaining what it does.

Isn't it also OK to set dtypes manually but make dtypesIfCUDA dynamic? That's the scenario @eellison was concerned about simplifying.

Reply:
> Let's add a comment explaining what it does.

Sure.

> Isn't it also OK to set dtypes manually but make dtypesIfCUDA dynamic? That's the scenario @eellison was concerned about simplifying.

I think as long as `dtypesIfCUDA` is dynamic, it should be ok.

Thanks!